### ISSUE TITLE
Support schedule events

### ISSUE THREAD

**Issue by ianlewis** [slsa-framework/github-actions-buildtypes#3]
https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#schedule

I don't recall the reason why they were excluded. Was there some info missing from those events that we cared about?

/cc @MarkLodato @laurentsimon @kpk47

**Comment by MarkLodato** [comment]
I suspect the reason was to avoid any ambiguity about which specific version was built. While the `GITHUB_SHA` and `GITHUB_REF` are set to the default branch, it seems conceivable that builds check out arbitrary branches during the workflow. Then again, it's possible to do that for the other events too.

I have no objection to adding it. It seems that the semantics are reasonably clear.

**Comment by ianlewis** [comment]
Right. I think `schedule` events are triggered on the default branch at HEAD so we would be presuming it was built from there but I think you're right it's not unlikely that someone would build from an arbitrary commit. However, from a technical perspective this is the same as how all other events work.

I just ran into this as part of tests, so there isn't an end user need for this yet. I'll let it stew a bit before sending a PR to add it.
